miniapps/elasticity/elasticity_driver.cpp

```diff
@@ -0,0 +1,177 @@
+
+using namespace std;
+using namespace mfem;
+
+/// This example only works in 3D.
+constexpr int dimension = 3;
+
+int main(int argc, char *argv[])
+{
+   Mpi::Init(argc, argv);
+
+   int order = 1;
+   const char *device_config = "cpu";
+   int diagpc_type = ElasticityDiagonalPreconditioner::Type::Diagonal;
+   int serial_refinement_levels = 0;
+   bool paraview = false;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree).");
+   args.AddOption(&device_config, "-d", "--device",
+                  "Device configuration string, see Device::Configure().");
+   args.AddOption(&diagpc_type, "-pc", "--pctype",
+                  "Select diagonal preconditioner type"
+                  " (0:Diagonal, 1:BlockDiagonal).");
+   args.AddOption(&serial_refinement_levels, "-rs", "--ref-serial",
+                  "Number of uniform refinements on the serial mesh.");
+   args.AddOption(&paraview, "-pv", "--paraview", "-no-pv",
+                  "--no-paraview",
+                  "Enable or disable ParaView DataCollection output.");
+   args.ParseCheck();
+
+   Device device(device_config);
+   if (Mpi::Root())
+   {
+      device.Print();
+   }
+
+   auto mesh =
+      Mesh::MakeCartesian3D(8, 2, 2, Element::HEXAHEDRON, 8.0, 1.0, 1.0);
+   if (mesh.Dimension() != dimension)
+   {
+      MFEM_ABORT("This example only works in 3D.");
+   }
+   mesh.EnsureNodes();
+
+   for (int l = 0; l < serial_refinement_levels; l++)
+   {
+      mesh.UniformRefinement();
+   }
+
+   ParMesh pmesh(MPI_COMM_WORLD, mesh);
+   mesh.Clear();
+
+   // Create the elasticity operator on the parallel mesh.
+   ElasticityOperator elasticity_op(pmesh, order);
+
+   // Create and set the material type. We define it's GradientType during
+   // instantiation.
+
+   // As seen in materials/gradient_type.hpp there is a choice of the
+   // GradientType with either
+   // * Symbolic
+   // * EnzymeFwd
+   // * EnzymeRev
+   // * FiniteDiff
+   // * DualNumbers
+   const NeoHookeanMaterial<dimension, GradientType::DualNumbers> material{};
+   elasticity_op.SetMaterial(material);
+
+   // Define all essential boundaries. In this specific example, this includes
+   // all fixed and statically displaced degrees of freedom on mesh entities in
+   // the defined attributes.
+   if (pmesh.bdr_attributes.Size())
+   {
+      Array<int> ess_attr(pmesh.bdr_attributes.Max());
+      ess_attr = 0;
+      ess_attr[4] = 1;
+      ess_attr[2] = 1;
+      elasticity_op.SetEssentialAttributes(ess_attr);
+   }
+
+   // Define all statically displaced mesh attributes. On these degrees of
+   // freedom (determined from the mesh attributes), a fixed displacement is
+   // prescribed.
+   if (pmesh.bdr_attributes.Size())
+   {
+      Array<int> displaced_attr(pmesh.bdr_attributes.Max());
+      displaced_attr = 0;
+      displaced_attr[2] = 1;
+      elasticity_op.SetDisplacedAttributes(displaced_attr);
```

Comment:
Same for the SetDisplacedAttributes method. They aren't displaced in the formulation.